src/index.ts

```diff
@@ -224,17 +230,15 @@ export const asyncWritable = <S extends Stores, T>(
     }
   };
 
-  const updateStoreValueThenWrite = async (updater: Updater<T>) => {
-    const currentValue = await loadDependenciesThenSet(loadAll);
-    return setStoreValueThenWrite(updater(currentValue));
-  };
+  const set = (newValue: T) => setStoreValueThenWrite(() => newValue);
```

Comment:
This is a little weird/hard to intuit. Basically set and update do the same thing, except the newValue is directly passed in for `set` and derived using the updater callback and the old value for `update.` 

So we can make this process completely equivalent by passing in an 'updater' for set that just always returns the new value.